bevy_lint/src/groups.rs

```diff
@@ -0,0 +1,66 @@
+use crate::LintGroup;
+use rustc_lint::Level;
+
+/// A group of deny-by-default lints that check for outright wrong or useless code.
+///
+/// These lints are carefully picked to be free of false-positives. You should avoid
+/// `#[allow(...)]`-ing these lints without a _very_ good reason.
+pub static CORRECTNESS: LintGroup = LintGroup {
+    name: "bevy::correctness",
+    level: Level::Deny,
+};
+
+/// A group similar to [`CORRECTNESS`] that checks for suspicious or usually wrong code.
+///
+/// As compared to [`CORRECTNESS`], it may be possible that the linted code may be written
+/// intentionally. Even still, you usually want to fix these lints instead of `#[allow(...)]`-ing
+/// them.
+pub static SUSPICIOUS: LintGroup = LintGroup {
+    name: "bevy::suspicious",
+    level: Level::Warn,
+};
+
+/// A group that offers suggestions on how to simplify your code.
+pub static COMPLEXITY: LintGroup = LintGroup {
+    name: "bevy::complexity",
+    level: Level::Warn,
+};
+
+/// A group that suggests how to increase the performance of your code.
+pub static PERFORMANCE: LintGroup = LintGroup {
+    name: "bevy::performance",
+    level: Level::Warn,
+};
+
+/// A group of lints that encourage idiomatic code.
+///
+/// These lints are opinionated and may be freely disabled if you disagree with their suggestions.
+pub static STYLE: LintGroup = LintGroup {
+    name: "bevy::style",
+    level: Level::Warn,
+};
+
+/// A group of lints that make the linter incredibly nit-picky.
+///
+/// If you enable this group, expect to liberally apply `#[allow(...)]` attributes throughout your
+/// code.
+pub static PEDANTIC: LintGroup = LintGroup {
+    name: "bevy::pedantic",
+    level: Level::Allow,
+};
+
+/// A group of opt-in lints that restrict you from writing certain code.
+///
+/// These are designed for scenarios where you want to increase the consistency of your code-base
+/// and reject certain patterns. They should not all be enabled at once, but instead specific lints
+/// should be individually enabled.
```

Comment:
I feel like the obvious question is, "if they shouldn't all be enabled at once, why are they in a group"? Or are we saying, this group is for categorising but not for actual use :thinking:

Reply:
I want every lint to be within a group so that it's clear what its general purpose is. These lint groups are just as important for documentation as they are for bulk-toggling.

So yes, this group is a category that's never intended to be named directly. (Clippy has the same things!)